lib/workers/repository/update/branch/auto-replace.spec.ts

```diff
@@ -1038,5 +1038,250 @@ describe('workers/repository/update/branch/auto-replace', () => {
         `
       );
     });
+
+    it('docker: updates with pinDigest enabled but no currentDigest value', async () => {
+      const dockerfile = codeBlock`
+        FROM ubuntu:18.04
+      `;
+      upgrade.manager = 'dockerfile';
+      upgrade.depName = 'ubuntu';
+      upgrade.currentValue = '18.04';
+      upgrade.currentDigest = undefined;
+      upgrade.depIndex = 0;
+      upgrade.pinDigests = true;
+      upgrade.updateType = 'replacement';
+      upgrade.replaceString = 'ubuntu:18.04';
+      upgrade.newName = 'alpine';
+      upgrade.newValue = '3.16';
+      upgrade.newDigest = 'sha256:p0o9i8u7z6t5r4e3w2q1';
+      upgrade.packageFile = 'Dockerfile';
+      const res = await doAutoReplace(upgrade, dockerfile, reuseExistingBranch);
+      expect(res).toBe(
+        codeBlock`
+          FROM alpine:3.16
```

Comment:
Shouldn't `pinDigests: true` lead to a digest even if there not one currently applied?
```suggestion
          FROM alpine:3.16@sha256:p0o9i8u7z6t5r4e3w2q1
```

Reply:
@secustor - thanks for the PR review.

I called out this behavior in this [comment above](https://github.com/renovatebot/renovate/pull/20317#issuecomment-1425112361) after feedback from @viceice [here](https://github.com/renovatebot/renovate/pull/20317#discussion_r1101824798)

Reply:
The missing `autoReplaceStringTemplate` which is added by the Dockerfile manager is the reason why . 
https://github.com/renovatebot/renovate/blob/90694bf398fea2f20c4892db48f61a6f420f8610/lib/modules/manager/dockerfile/extract.ts#L33

Reply:
Thanks @secustor - will update the test cases later today

Reply:
Ok, I've double confirmed what I mentioned above , adding `upgrade.autoReplaceStringTemplate=
        '{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}';` has no bearing with replacements because of how the [logic is inside doAutoReplace](https://github.com/renovatebot/renovate/blob/8af82990ee38f4e35071af01c8d124f401fa2bc1/lib/workers/repository/update/branch/auto-replace.ts#L196)